bdns/dns.go

```diff
@@ -177,6 +249,14 @@ func NewTestDNSResolverImpl(readTimeout time.Duration, servers []string, stats m
 	return resolver
 }
 
+// LookupIPV6 controls whether the LookupHost function resolves AAAA records.
```

Comment:
@rolandshoemaker: No, the plan is that it's a typical feature flag, goes away after it's flipped on.

@riking and I talked over this in person a bit on Monday. There are three possible ways to handle this:

1. Exposed bool, it's up to the calling code to set this only once, and before any other DNS code runs
2. Setter with a lock, can be called any time because it's thread safe.
3. Builder pattern: DNSBuilder.WithIPv6().Build().
4. Make the bool a parameter of the constructor.

This isn't any of those, really. It's a setter, but with no lock. It has a return value that allows chaining (like a builder pattern), but since the return value is a `DNSResolver`, not a builder, it's not really the builder pattern. The critical distinction is that you are constructing the object and then modifying it, rather than setting the parameters with which you want to construct it, then constructing it.

We should properly do one of the four options. I originally recommended against (1), but @rolandshoemaker does have a good point that we'll (hopefully) be deleting the code soon. So I'd be fine with it.